### PR TITLE
Maintenance

### PR DESCRIPTION
**Pull request by dvc94ch**
Does a bit of maintenance. Maybe you could give me some pointers as to what the status is and what needs to be done. I'll have to read the specs more carefully, I'm not a cryptographer...

### CODE REVIEW

File: src/apis.rs [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-
 use std::error;
 use std::fmt;
 
-use asymmetric::DH_SIZE;
-use config::Config;
-use disco::{DiscoReadError, DiscoWriteError, HandshakeState, KEY_SIZE};
+use crate::asymmetric::DH_SIZE;
+use crate::config::Config;
+use crate::disco::{DiscoReadError, DiscoWriteError, HandshakeState, KEY_SIZE};
```
Comment (mimoo) [inline src/apis.rs:8]: why adding crate here?

File: src/asymmetric.rs [hunk 2]
```diff
@@ -9,14 +8,15 @@ pub const DH_SIZE: usize = 32;
 /// Contains a private and a public part. It can be generated via the `gen` or `from_priv_key`
 /// functions. The public part can also be extracted via the `pub_key_bytes` function.
 pub struct KeyPair {
-    pub(crate) pub_key: MontgomeryPoint,
-    priv_key: [u8; 32],
+    pub(crate) pub_key: PublicKey,
+    priv_key: StaticSecret,
```
Comment (mimoo) [inline src/asymmetric.rs:12]: 👍

File: src/disco.rs [hunk 3]
```diff
@@ -72,16 +74,14 @@ impl SymmetricState {
     /// is keyed and the MAC does not pass verification.
     fn decrypt_and_hash(&mut self, mut bytes: Vec<u8>) -> Result<Vec<u8>, AuthError> {
         if self.is_keyed {
-            if bytes.len() < TAG_SIZE {
-                panic!("decrypt_and_hash input too short");
-            }
+            assert!(bytes.len() >= TAG_SIZE);
```
Comment (mimoo) [inline src/disco.rs]: you probably don't want to panic here

File: src/lib.rs [hunk 4]
```diff
@@ -1,9 +1,3 @@
-extern crate curve25519_dalek;
-extern crate rand;
-extern crate strobe_rs;
-extern crate take_mut;
-extern crate x25519_dalek;
-
```
Comment (mimoo) [inline src/lib.rs:6]: 👍

File: src/symmetric.rs [hunk 5]
```diff
@@ -151,30 +146,28 @@ pub fn hash(input_data: Vec<u8>, output_len: usize) -> Vec<u8> {
...
-                security consequences")
-    }
-
-    let mut s = Strobe::new("DiscoKDF".as_bytes().to_vec(), SecParam::B128);
-    s.ad(input_key, None, false);
-    s.prf(output_len, None, false)
+pub fn derive_keys(input_key: &[u8], output_len: usize) -> Vec<u8> {
+    assert!(input_key.len() >= 16);
+
+    let mut s = Strobe::new(b"DiscoKDF", SecParam::B128);
```
Comment (mimoo) [inline src/symmetric.rs:152]: this is sweet!

File: src/apis.rs (hunk 1 above)
Comment (dvc94ch) [reply src/apis.rs:8]: it's how you reference the current crate in the 2018 edition. In the 2015 edition it was implicit.

- [0] https://doc.rust-lang.org/nightly/edition-guide/rust-2018/module-system/path-clarity.html

Comment (mimoo) [reply src/apis.rs:8]: oh right!

**Comment by mimoo** [comment]
LGTM! I'll let @rozbb take a look before it gets merged though :)

File: src/disco.rs [hunk 6]
```diff
@@ -74,7 +74,9 @@ impl SymmetricState {
     /// is keyed and the MAC does not pass verification.
     fn decrypt_and_hash(&mut self, mut bytes: Vec<u8>) -> Result<Vec<u8>, AuthError> {
         if self.is_keyed {
-            assert!(bytes.len() >= TAG_SIZE);
+            if bytes.len() < TAG_SIZE {
+                return Err(AuthError);
+            }
```
Comment (mimoo) [inline src/disco.rs:79]: awesome!

**Comment by dvc94ch** [comment]
it is now at https://github.com/rust-p2p/disco

**Pull request closed without merging**